### PR TITLE
Add button type property in mwc-button

### PR DESCRIPTION
**Pull request by jpnathan**
Issue: Button type submit #205

### CODE REVIEW

**Comment by azakus** [comment]
Related to #289

File: packages/button/src/mwc-button-base.ts [hunk 1]
```diff
@@ -32,6 +32,8 @@ export class ButtonBase extends LitElement {
 
   @property({type: Boolean}) trailingIcon = false;
 
+  @property({type: String}) type = 'submit';
```
Comment (fernandopasik) [inline packages/button/src/mwc-button-base.ts:35]: I think the in the html spec the default type is `button`

Also might be good to add a typescript string literal type for this
`type ButtonType = 'button' | 'submit' | 'reset'`

resulting in 
`@property({type: String}) type: ButtonType = 'button';`

File: packages/button/src/mwc-button-base.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-import {HTMLElementWithRipple} from '@material/mwc-base/form-element';
+import {HTMLElementWithRipple } from '@material/mwc-base/form-element';
```
Comment (fernandopasik) [inline packages/button/src/mwc-button-base.ts:17]: Please remove this unnecessary white space

**Comment by fernandopasik** [comment]
> Related to #289

@dfreedm I know that is related to the form submit ticket, but why should that block adding this feature?

File: packages/button/src/mwc-button-base.ts (hunk 1 above)
Comment (salomvary) [reply packages/button/src/mwc-button-base.ts:35]: FYI the default type in HTML is `submit`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type

Comment (fernandopasik) [reply packages/button/src/mwc-button-base.ts:35]: sorry about that, yes `submit` is the default

**Comment by asyncLiz** [comment]
What does adding this feature without custom element form submission accomplish? 

An `<mwc-button type="submit">` inside a `<form>` element will not submit it. I'd be cautious about adding `type="submit"` which could imply to users that form submission works when it doesn't.

**Comment by fernandopasik** [comment]
Makes sense @asyncLiz, thanks. We'll need to wait on the result of #289

**Comment by asyncLiz** [comment]
Closing as obsolete

**Comment by fernandopasik** [comment]
The type attribute wasn't added to button? Is it resolved in some other way?

**Comment by asyncLiz** [comment]
Sorry about that! Was going through and cleaning up a lot of old issues :)

**Comment by fernandopasik** [comment]
would it make sense to rebase this PR and try to get it ready to merge?

**Comment by asyncLiz** [comment]
> would it make sense to rebase this PR and try to get it ready to merge?

If you can rebase it for MWC (skip M3), I'll review it :)

**Comment by asyncLiz** [comment]
Obsolete with M3. See https://github.com/material-components/material-web/issues/3941 for upcoming support

**Pull request closed without merging**